daemon/execdriver/native/seccomp_default.go

```diff
@@ -0,0 +1,319 @@
+			// They're gated by CAP_SYS_NICE,
+			// which we do not retain by default in containers.
+			Name:   "mbind",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Terrifying syscalls that modify kernel memory and NUMA settings.
+			// They're gated by CAP_SYS_NICE,
+			// which we do not retain by default in containers.
+			Name:   "migrate_pages",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Old syscall only used in 16-bit code,
+			// and a potential information leak
+			Name:   "modify_ldt",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Deny mount
+			Name:   "mount",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Terrifying syscalls that modify kernel memory and NUMA settings.
+			// They're gated by CAP_SYS_NICE,
+			// which we do not retain by default in containers.
+			Name:   "move_pages",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Deny interaction with the kernel nfs daemon
+			Name:   "nfsservctl",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Cause of an old container breakout,
+			// might as well restrict it to be on the safe side
+			Name:   "open_by_handle_at",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Tracing/profiling syscalls,
+			// which could leak a lot of information on the host
+			Name:   "perf_event_open",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Prevent container from enabling BSD emulation.
+			// Not inherently dangerous, but poorly tested,
+			// potential for a lot of kernel vulns in this.
+			Name:   "personality",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Deny pivot_root
+			Name:   "pivot_root",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Already blocked by dropping CAP_PTRACE
+			Name:   "ptrace",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Deny manipulation and functions on kernel modules.
+			Name:   "query_module",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Quota and Accounting syscalls which could let containers
+			// disable their own resource limits or process accounting
+			Name:   "quotactl",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Probably a bad idea to let containers reboot the host
+			Name:   "reboot",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Probably a bad idea to let containers restart
+			Name:   "restart_syscall",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Prevent containers from using the kernel keyring,
+			// which is not namespaced
+			Name:   "request_key",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// meta, deny seccomp
+			Name:   "seccomp",
```

Comment:
I feel like if an app is setting its own seccomp rules the Dev can make their own custom seccomp profile and should... Who knows if down the line this will have a bypass, also you can't load a new apparmor profile inside a container unprivileged even tho this is different and can only restrict it more I don't trust it


On Dec 29, 2015, 07:23 -0800, Aleksa Sarai<notifications@github.com>, wrote:
>  
> Indaemon/execdriver/native/seccomp_default.go(https://github.com/docker/docker/pull/18780#discussion_r48546597):
>  
> >+     {>+         // Probably a bad idea to let containers restart>+         Name:  "restart_syscall",>+         Action: configs.Errno,>+         Args:  []*configs.Arg{},>+     },>+     {>+         // Prevent containers from using the kernel keyring,>+         // which is not namespaced>+         Name:  "request_key",>+         Action: configs.Errno,>+         Args:  []*configs.Arg{},>+     },>+     {>+         // meta, deny seccomp>+         Name:  "seccomp",
>  
> @jfrazelle(https://github.com/jfrazelle)AFAICS, this is not necessary. Seccomp does not allow a process running under a seccomp context to remove restrictions (otherwise it would be a pointless security feature). A Docker container could only make its seccomp context more restrictive (which I'm not sure if we should be blocking).
>  
>  
> —
> Reply to this email directly orview it on GitHub(https://github.com/docker/docker/pull/18780/files#r48546597).
>  
>  
>